vkdgen.py

```diff
@@ -133,12 +133,12 @@ def __init__(self, errFile=sys.stderr, warnFile=sys.stderr, diagFile=sys.stderr)
 		self.functionVars = ""
 		self.opaqueStruct = set()
 		self.surfaceExtensions = {
-			"// VK_KHR_android_surface" : "VK_USE_PLATFORM_ANDROID_KHR",
-			"// VK_KHR_mir_surface"     : "VK_USE_PLATFORM_MIR_KHR",
-			"// VK_KHR_wayland_surface" : "VK_USE_PLATFORM_WAYLAND_KHR",
-			"// VK_KHR_win32_surface"   : "VK_USE_PLATFORM_WIN32_KHR",
-			"// VK_KHR_xlib_surface"    : "VK_USE_PLATFORM_XLIB_KHR",
-			"// VK_KHR_xcb_surface"	    : "VK_USE_PLATFORM_XCB_KHR",
+			"// VK_KHR_android_surface" : ["VK_USE_PLATFORM_ANDROID_KHR",	"public import android.native_window;\n"],
+			"// VK_KHR_mir_surface"     : ["VK_USE_PLATFORM_MIR_KHR",		"public import mir_toolkit.client_types;\n"],
+			"// VK_KHR_wayland_surface" : ["VK_USE_PLATFORM_WAYLAND_KHR",	"public import wayland_client;\n"],
+			"// VK_KHR_win32_surface"   : ["VK_USE_PLATFORM_WIN32_KHR",		"public import core.sys.windows.windows;\n"],
+			"// VK_KHR_xlib_surface"    : ["VK_USE_PLATFORM_XLIB_KHR",		"public import X11.Xlib;\n"],
+			"// VK_KHR_xcb_surface"	    : ["VK_USE_PLATFORM_XCB_KHR",		"public import xcb.xcb;\n"],
 		}

```

Comment:
I get them from vulkan.h. Those are the D module names of the corresponding C include headers. Now to the idea.
d-vulkan folder structure:
`source/dvulkan/...`

if someone wants to use e.g. xcb for creating a window, all he needs to do is to add:
`source/xcb/xcb.d`

to the folder structure above. Copy it there and let git ignore it. Its up to the user to get those bindings from whatever-where. Finally VK_USE_PLATFORM_XCB_KHR must be specified and the xcb would be imported and used.

Please note that till now the vulkan.d and vulkan.h are almost similar in structure and in functionality (personally I feel this is quite important)

Now a new variant came up which enables the usage of dub bindings. Nothing has to be changed in vulkan.d, only d-vulkan/erupted.dub would change.

All the stuff is explained on the erupted README.md. Did you have a chance to read it?